pkg/cimfs/cim_test.go

```diff
@@ -60,6 +84,99 @@ func createCimFileUtil(c *CimFsWriter, fileTuple tuple) error {
 	return nil
 }
 
+// openNewCIM creates a new CIM and returns a writer to that CIM.  The caller MUST close
+// the writer.
+func openNewCIM(t *testing.T, newCIM testCIM) *CimFsWriter {
+	t.Helper()
+
+	var (
+		writer *CimFsWriter
+		err    error
+	)
+
+	switch val := newCIM.(type) {
+	case *testForkedCIM:
+		writer, err = Create(val.imageDir, val.parentName, val.imageName)
+	case *testBlockCIM:
+		writer, err = CreateBlockCIM(val.BlockPath, val.CimName, val.Type)
+	}
+	if err != nil {
+		t.Fatalf("failed while creating a cim: %s", err)
+	}
+	t.Cleanup(func() {
+		writer.Close()
+		// add 3 second sleep before test cleanup remove the cim directory
+		// otherwise, that removal fails due to some handles still being open
+		time.Sleep(3 * time.Second)
+	})
+	return writer
+}
+
+// compareContent takes in path to a directory (which is usually a volume at which a CIM is
+// mounted) and ensures that every file/directory in the `testContents` shows up exactly
+// as it is under that directory.
+func compareContent(t *testing.T, root string, testContents []tuple) {
+	t.Helper()
+
+	for _, ft := range testContents {
+		if ft.isDir {
+			_, err := os.Stat(filepath.Join(root, ft.filepath))
+			if err != nil {
+				t.Fatalf("stat directory %s from cim: %s", ft.filepath, err)
+			}
+		} else {
+			f, err := os.Open(filepath.Join(root, ft.filepath))
+			if err != nil {
+				t.Fatalf("open file %s: %s", filepath.Join(root, ft.filepath), err)
+			}
+			defer f.Close()
+
+			fileContents := make([]byte, len(ft.fileContents))
+
+			// it is a file - read contents
+			rc, err := f.Read(fileContents)
```

Comment:
`io.(Reader).Read` is not guaranteed to read the full buffer. I would use `io.ReadAll` or `os.ReadFile` instead.